llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -3100,9 +3100,8 @@ void InstCombinerImpl::handleUnreachableFrom(
     MadeIRChange = true;
   }
 
-  // RemoveDIs: to match behaviour in dbg.value mode, drop debug-info on
-  // terminator too.
-  BB->getTerminator()->dropDbgValues();
+  // if terminator refer instruction, transform it into poison.
+  handleUnreachableTerminator(BB->getTerminator());
```

Comment:
```suggestion
  MadeIRChange |= handleUnreachableTerminator(BB->getTerminator());
```
and return whether a change was made from that function.